### PR TITLE
fix: store partial_filter_selector in ddoc

### LINKED ISSUE #8522: pouchdb-find createIndex doesn't write partial_filter_selector
### Issue
The `partial_filter_selector` field is not stored when pouchdb-find's `createIndex` is used on a local db.

### Info
- Environment: Node.js
- Platform: FF
- Adapter: idb/indexeddb/memory/leveldb/etc.
- Server: none

### Reproduce
```javascript
pouchdb.createIndex(index: {
    fields: ['hello'],
    partial_filter_selector: { type: 'x' }
  },
  ddoc: 'test-partial',
  name: 'type-x'
})
  .then(result => pouchdb.get('_design/test-partial'))
  .then(ddoc => {
    console.log(ddoc.views['type-x'].map.partial_filter_selector) // logs undefined
  })
```

### PR DESCRIPTION
**Pull request by Terreii**
This pull request fixes #8522.

### CODE REVIEW

**Comment by Terreii** [comment]
Now only indexeddb adapter fails. I will look into it, on Sunday.

**Comment by Terreii** [comment]
OK, indexeddb adapter implements its own [query interface](https://github.com/pouchdb/pouchdb/blob/9519c748c48cbac41451039765c758e5987bbc57/packages/node_modules/pouchdb-adapter-indexeddb/src/index.js#L129) (which is used by find).
It only uses the [fields to find an index](https://github.com/pouchdb/pouchdb/blob/9519c748c48cbac41451039765c758e5987bbc57/packages/node_modules/pouchdb-adapter-indexeddb/src/find.js#L139). Those indexes are [native indexes](https://github.com/pouchdb/pouchdb/blob/9519c748c48cbac41451039765c758e5987bbc57/packages/node_modules/pouchdb-adapter-indexeddb/src/setup.js#L43), and they only [don't support filtering](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/createIndex).
So I might have to create a fields array with the `partial_filter_selector` included and combine it with the user-provided fields-array.

**Comment by Terreii** [comment]
Sorry for the long silence.
I fixed it by adding to `_customFindAbstractMapper` a third argument: The standard/original `findAbstractMapper`. And then in the _indexeddb adapter_, I added a check if a view has a `partial_filter_selector`. And if so use the standard mapper (a fallback).

**Comment by Terreii** [comment]
@pouchdb OK Pull request is refactored, squashed, and all tests pass!
I think it is ready to review/merge.

**Comment by Terreii** [comment]
@AlbaHerrerias I think this pull request is ready for merging.

**Review by garbados — APPROVED** [review]
LGTM!

**Comment by janl** [comment]
> In other words: find indexes with a partial_filter_selector are on indexeddb adapter db (like on all other adapters) just like map-reduce-views.

We should flag that in the docs then, that using partial filters on `indexeddb` will negate any of the perf benefits of native indexes (arguable one of the two prime reasons for `indexeddb`’s existence :)

**Comment by Terreii** [comment]
I did add a warning.
Once this pr is merged, I will open an Issue, for finding a way to fix performance in the indexeddb-adapter. Which will be complicated, a reason why I didn't do it in a bug fix.

**Comment by Terreii** [comment]
OK, I think __test.bulk_docs.js-http #3062 bulkDocs with staggered seqs__ is on pouchdb-server unstable.

File: docs/_includes/api/create_index.html [hunk 1]
```diff
@@ -212,6 +212,17 @@
 
 If you want to index a subset of the documents in the database, you can use `partial_filter_selector`. This has the same syntax as the selector you'd pass to `find()`, and only documents matching the selector will be included in the index.
 
+{% include alert/start.html variant="warning"%}
+{% markdown %}
+
+**When using _pouchdb-adapter-indexeddb:_** On the `indexeddb`-adapter, indexes with
+a`partial_filter_selector` will fallback to using [map-reduce](#query_database),
```
Comment (janl) [inline docs/_includes/api/create_index.html:219]:
```suggestion
a`partial_filter_selector` will fall back to using [map-reduce](#query_database),
```

File: docs/_includes/api/create_index.html [hunk 2]
```diff
@@ -212,6 +212,17 @@
...
+instead of the native indexes. Negating all index performance gains from the
```
Comment (janl) [inline docs/_includes/api/create_index.html:220]:
```suggestion
instead of the native indexes, negating all index performance gains from the
```

**Review by janl — CHANGES_REQUESTED** [review]

**Comment by janl** [comment]
>  for finding a way to fix performance in the indexeddb-adapter.

I’d be curious to know how you think this can be fixed :)

**Comment by Terreii** [comment]
> > for finding a way to fix performance in the indexeddb-adapter.
> 
> I’d be curious to know how you think this can be fixed :)

At the moment only by trying to translate filters into indexes, and some into the selectors on query-time. With a fall back to map-reduce if that doesn't work.

But the more I think about it, the more I remember why I droped that path in this fix.

...or it may could act as an use case for a proposal to add partial indexes to IndexedDB indexes (after all SQLite [has them](https://www.sqlite.org/partialindex.html), too).

**Comment by janl** [comment]
I like your thinking :D

**Pull request merged**